api/src/gmsa_service.cpp

```diff
@@ -1149,3 +1153,144 @@ int parse_cred_spec( std::string credspec_data, creds_fetcher::krb_ticket_info*
 
     return 0;
 }
+
+
+/**
+ * ProcessCredSpecFile - Processes a provided credential spec file
+ * @param krb_files_dir - Kerberos TGT directory
+ * @param credspec_filepath - Path to credential spec file produced by DC
+ * @param cf_logger - log to systemd daemon
+ * @return - return 0 on success
+ */
+int ProcessCredSpecFile(std::string krb_files_dir, std::string credspec_filepath, creds_fetcher::CF_logger& cf_logger) {
+    std::string lease_id = "credspec";
```

Comment:
@smhmhmd @saikiranakula-amzn I have addressed this also and will have the behavior change in the updated PR in a day or two

Reply:
I have changed the behavior to address the lease id being optionally set when using the environment variable CF_CRED_SPEC_FILE.  The environment variable takes the path to the cred spec file, i.e. CF_CRED_SPEC_FILE=/var/my-cred-spec.json   This produces a default leaseid of 'credspec'.  Optionally the CF_CRED_SPEC_FILE=/var/my-cred-spec.json:myLeaseId   In this case the lease id is defined following the ':' in the path of the cred spec file.  I chose this method over an argument because I wanted the env path to be coupled in definition to a lease id if its defined in the env variable.